src/engraving/libmscore/score.cpp

```diff
@@ -5417,9 +5417,22 @@ std::set<staff_idx_t> Score::staffIdsFromRange(const track_idx_t trackFrom, cons
             continue;
         }
 
-        std::set<staff_idx_t> staffIdxList = part->staveIdxList();
+        const std::vector<Staff*>& staves = part->staves();
 
-        result.insert(staffIdxList.cbegin(), staffIdxList.cend());
+        for (staff_idx_t idx = 0; idx < staves.size(); ++idx) {
+            const Staff* staff = staves.at(idx);
+            if (!staff) {
+                continue;
+            }
+
+            if (staffAccepted) {
+                if (staffAccepted(*staff)) {
+                    result.insert(idx);
+                }
+            } else {
+                result.insert(idx);
+            }
+        }
```

Comment:
I'm afraid this doesn't work well in scores with multiple `Part`s. Now, `idx` starts from zero _for each part_, while `Part::staveIdxList()` returned the indexes of the staves _in the whole score_.

Reply:
Thank you for finding this bug! The reason I don't want to use Part::staveIdxList() here is that it uses staff->idx(), which has linear complexity, which makes this algorithm very inefficient. I've fixed the implementation of this method and added a test